pkgs/tools/networking/sslsplit/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchFromGitHub, openssl, libevent, libpcap, libnet, zlib }:
+
+stdenv.mkDerivation rec {
+  pname = "sslsplit";
+  version = "0.5.4";
+
+  src = fetchFromGitHub {
+    owner = "droe";
+    repo = pname;
+    rev = version;
+    sha256 = "1zl6058ls591w3qizzgns8nb9d0n7mwygda2l3wis9jlfzj92mis";
+  };
+
+  buildInputs = [ openssl libevent libpcap libnet zlib ];
+
+  makeFlags = [
+    "DESTDIR=$(out)"
```

Comment:
Using `DESTDIR=$(out) PREFIX=` is not a good idea because if the source code is hardcoding `PREFIX "foo"`, it will try to open `foo` instead of `$out/foo`.

Instead you should apply the following patch:

```patch
diff --git a/pkgs/tools/networking/sslsplit/default.nix b/pkgs/tools/networking/sslsplit/default.nix
index 087ec7c179d..bcf9b0faa34 100644
--- a/pkgs/tools/networking/sslsplit/default.nix
+++ b/pkgs/tools/networking/sslsplit/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub, openssl, libevent, libpcap, libnet, zlib }:
+{ stdenv, fetchFromGitHub, fetchpatch, openssl, libevent, libpcap, libnet, zlib }:
 
 stdenv.mkDerivation rec {
   pname = "sslsplit";
@@ -11,11 +11,19 @@ stdenv.mkDerivation rec {
     sha256 = "1zl6058ls591w3qizzgns8nb9d0n7mwygda2l3wis9jlfzj92mis";
   };
 
+  patches = [
+    # Do not try to create /var directory
+    (fetchpatch {
+      url = https://github.com/droe/sslsplit/commit/40c22d725668fe44a9acba0358e1e08298a75b02.patch;
+      sha256 = "0z1vhfr50x8mf7k5pxkn2dxws5k0r8cdz715p241azh2773ms1sn";
+      excludes = [ "NEWS.md" ];
+    })
+  ];
+
   buildInputs = [ openssl libevent libpcap libnet zlib ];
 
   makeFlags = [
-    "DESTDIR=$(out)"
-    "PREFIX="
+    "PREFIX=${placeholder "out"}"
     "OPENSSL_BASE=${openssl.dev}"
     "LIBEVENT_BASE=${libevent.dev}"
     "LIBPCAP_BASE=${libpcap}"
```

Reply:
You concern is valid. But many environment variables have to change for this package. Not just `PREFIX`. I ran into problems before using `DESTDIR=$(out) PREFIX=`.

Reply:
I opened an issue about thus in nixpkgs: https://github.com/NixOS/nixpkgs/issues/65718

What other problems have you seen?

Reply:
I don't quite remember. Sorry. I will try to rebuild it.

Reply:
Here is the install target from of this package's 0.5.4 release's GNUMakefile,

```
install: $(TARGET) $(TARGET).conf $(TARGET).1 $(TARGET).conf.5
        test -d $(DESTDIR)$(PREFIX)/bin || $(MKDIR) -p $(DESTDIR)$(PREFIX)/bin
        test -d $(DESTDIR)$(PREFIX)/$(TARGET) || \
                $(MKDIR) -p $(DESTDIR)$(PREFIX)/sslsplit
        test -d $(DESTDIR)$(PREFIX)/$(MANDIR)/man1 || \
                $(MKDIR) -p $(DESTDIR)$(PREFIX)/$(MANDIR)/man1
        test -d $(DESTDIR)$(PREFIX)/$(MANDIR)/man5 || \
                $(MKDIR) -p $(DESTDIR)$(PREFIX)/$(MANDIR)/man5
        test -d $(DESTDIR)/var/log/$(TARGET) || \
                $(MKDIR) -p $(DESTDIR)/var/log/$(TARGET)
        test -d $(DESTDIR)/var/run/$(TARGET) || \
                $(MKDIR) -p $(DESTDIR)/var/run/$(TARGET)
        $(INSTALL) $(BINOWNERFLAGS) -m $(BINMODE) \
                $(TARGET) $(DESTDIR)$(PREFIX)/bin/
        $(INSTALL) $(CNFOWNERFLAGS) -m $(CNFMODE) \
                $(TARGET).conf \
                $(DESTDIR)$(PREFIX)/$(TARGET)/$(TARGET).conf.sample
        $(INSTALL) $(MANOWNERFLAGS) -m $(MANMODE) \
                $(TARGET).1 $(DESTDIR)$(PREFIX)/$(MANDIR)/man1/
        $(INSTALL) $(MANOWNERFLAGS) -m $(MANMODE) \
                $(TARGET).conf.5 $(DESTDIR)$(PREFIX)/$(MANDIR)/man5/

```

`$(DESTDIR)/var/log/$(TARGET)` path is hardcoded. So I will need to set `$DESDIR` anyway. That's why I only set `$DESTDIR`.

This commit https://github.com/droe/sslsplit/commit
/597c1a2afc78ded0c3313b8326bd031075c150d3#diff-b0fa74059a31d3efac99fea6d20ca25a has fixed that. But it is still not been tagged for release yet.

Reply:
You can use the patch as described above, which is IMO a cleaner solution.

Reply:
I have created a patch from current HEAD and modified the nix file.